### PR TITLE
Polyhedron demo: enhance point set

### PR DESCRIPTION
**Pull request by sgiraudot**
As I already discussed it with @lrineau, I would like to add the `Point_set_3` structure (currently reviewed as package for CGAL) in the Polyhedron demo as soon as possible:

* It changes _a lot_ of files in the demo, and as the demo changes very frequently, I always have tons of conflicts when refreshing the branch from master. As the feature might take weeks or months, this means that many more conflicts will probably happen later
* Although the API is still likely to change (as the feature is under review), the work needed to propagate such API changes will be much easier than the work to solve conflicts with the current API
* Even if the feature ends up not being accepted by the editorial board, it's still something that makes sense in the demo (colored point sets, better PLY reader/writer, reduced memory weight, etc.)
* Including the feature only in the Polyhedron demo is very straightforward, just 3 headers moved in the `Polyhedron/demo/Polyhedron/include` directory. These headers will just have to be removed when/if the actual package `Point_set_3` arrives in master

Note that this PR also modifies some files in `Surface_mesh` and `Point_set_processing` (for example, moving the `Properties` of Surface mesh out of the class and into its own namespace), but the public API does not change. Examples and tests of these packages were tested locally.

### CODE REVIEW

**Comment by lrineau** [comment]
After the bunch of merges into `master`, this PR has conflicts with `master`. Please rebase the branch if possible.

**Comment by sgiraudot** [comment]
Conflicts are fixed.

**Comment by lrineau** [comment]
@sgiraudot @maxGimeno 
I'd like to have this PR tested in `integration`, but there are conflicts with #1323 (already in `integration`), in `Polyhedron/demo/Polyhedron/Scene_points_with_normal_item.cpp`, that I do not know how to resolve:

```diff
diff --cc Polyhedron/demo/Polyhedron/Scene_points_with_normal_item.cpp
index b8179de,b73ac39..0000000
--- a/Polyhedron/demo/Polyhedron/Scene_points_with_normal_item.cpp
+++ b/Polyhedron/demo/Polyhedron/Scene_points_with_normal_item.cpp
@@@ -619,7 -654,7 +662,11 @@@ void Scene_points_with_normal_item::dra
        ratio_displayed = 3 * limit_fast_drawing / (double)(d->nb_points + d->nb_selected_points);
  
      vaos[Scene_points_with_normal_item_priv::ThePoints]->bind();
++<<<<<<< HEAD
 +    if(has_normals() && renderingMode() == ShadedPoints)
++=======
+     if(has_normals() && !(d->m_points->has_colors()))
++>>>>>>> sgiraudot/Polyhedron_demo-Enhance_point_set-GF
      {
        d->program=getShaderProgram(PROGRAM_WITH_LIGHT);
        attribBuffers(viewer,PROGRAM_WITH_LIGHT);
@@@ -630,17 -665,15 +677,26 @@@
        attribBuffers(viewer,PROGRAM_NO_SELECTION);
      }
      d->program->bind();
++<<<<<<< HEAD
 +    d->program->setAttributeValue("colors", this->color());
 +    if(renderingMode() != ShadedPoints)
 +      d->program->setAttributeValue("normals", QVector3D(0,0,0));
 +
++=======
+     if (!(d->m_points->has_colors()))
+       d->program->setAttributeValue("colors", this->color());
++>>>>>>> sgiraudot/Polyhedron_demo-Enhance_point_set-GF
      viewer->glDrawArrays(GL_POINTS, 0,
                           static_cast<GLsizei>(((std::size_t)(ratio_displayed * d->nb_points)/3)));
      vaos[Scene_points_with_normal_item_priv::ThePoints]->release();
      d->program->release();
  
      vaos[Scene_points_with_normal_item_priv::Selected_points]->bind();
++<<<<<<< HEAD
 +    if(has_normals() && renderingMode() == ShadedPoints)
++=======
+     if(has_normals() && !(d->m_points->has_colors()))
++>>>>>>> sgiraudot/Polyhedron_demo-Enhance_point_set-GF
      {
        d->program=getShaderProgram(PROGRAM_WITH_LIGHT);
        attribBuffers(viewer,PROGRAM_WITH_LIGHT);
```

**Comment by sgiraudot** [comment]
Okay, so if I understand correctly, there will now be a way to choose if lighting is enabled (for the moment, it is automatically enabled if there are normal vectors).

From what I saw, lighting and colors do not really go well together (especially if they are real-world colors). But then again, if the colors are artificial (for example one color = one label), it could be interesting to have lighting with colors.

So in my opinion, we can keep the tests from the other branch:
```if(has_normals() && renderingMode() == ShadedPoints)```
and not
```if(has_normals() && !(d->m_points->has_colors()))```

The conflict in the middle:

```
++<<<<<<< HEAD
 +    d->program->setAttributeValue("colors", this->color());
 +    if(renderingMode() != ShadedPoints)
 +      d->program->setAttributeValue("normals", QVector3D(0,0,0));
 +
++=======
+     if (!(d->m_points->has_colors()))
+       d->program->setAttributeValue("colors", this->color());
++>>>>>>> sgiraudot/Polyhedron_demo-Enhance_point_set-GF
```

should IMO be solved like that:
```
    if (!(d->m_points->has_colors()))
      d->program->setAttributeValue("colors", this->color());
    if(renderingMode() != ShadedPoints)
      d->program->setAttributeValue("normals", QVector3D(0,0,0));
```

Do you want me to do it? (merging this branch into integration)

**Comment by lrineau** [comment]
@sloriot:
> Do you want me to do it? (merging this branch into integration)

Yes, that would be a good idea, so that I do not create errors.

**Comment by maxGimeno** [comment]
I agree with @sgiraudot for both the conflict resolutions. The lighting of the point_set only depends of the ShadedPoints rendering mode, not the colors, so the first test should be kept, and both tests should be kept in the second conflict.

**Comment by sgiraudot** [comment]
Okay, I tested it and it works, so it's merged into integration.

**Comment by lrineau** [comment]
This PR breaks the testsuite of BGL:
```
[  4%] Building CXX object examples/BGL_surface_mesh/CMakeFiles/connected_components.dir/connected_components.cpp.o
/home/lrineau/Git/cgal/BGL/examples/BGL_surface_mesh/connected_components.cpp: In function ‘int main(int, char**)’:
/home/lrineau/Git/cgal/BGL/examples/BGL_surface_mesh/connected_components.cpp:21:3: error: ‘Property_map’ is not a member of ‘Mesh {aka CGAL::Surface_mesh<CGAL::Point_3<CGAL::Simple_cartesian<double> > >}’
   Mesh::Property_map<vertex_descriptor,int> ccmap;
   ^~~~
/home/lrineau/Git/cgal/BGL/examples/BGL_surface_mesh/connected_components.cpp:21:39: error: expected primary-expression before ‘,’ token
   Mesh::Property_map<vertex_descriptor,int> ccmap;
                                       ^
/home/lrineau/Git/cgal/BGL/examples/BGL_surface_mesh/connected_components.cpp:21:40: error: expected primary-expression before ‘int’
   Mesh::Property_map<vertex_descriptor,int> ccmap;
                                        ^~~
/home/lrineau/Git/cgal/BGL/examples/BGL_surface_mesh/connected_components.cpp:22:3: error: ‘ccmap’ was not declared in this scope
   ccmap = sm.add_property_map<vertex_descriptor,int>("v:CC").first;
   ^~~~~
/home/lrineau/Git/cgal/BGL/examples/BGL_surface_mesh/connected_components.cpp:24:43: error: ‘connected_components’ was not declared in this scope
   int num = connected_components(sm, ccmap);
                                           ^
```

@sgiraudot  Please fix that today, or tell if you cannot, so that I can unmerge the PR from `cgal-dev/integration`.

**Comment by lrineau** [comment]
Note: You should add backward compatibility. The unmodified example should still compile.

**Comment by sgiraudot** [comment]
Okay, I fixed it and cherry-picked the commit in integration.

Backward compatibility is not easy in that case, because it's a template type I cannot just do `CGAL::Properties::Property_map<> Property_map<>` inside `Surface_mesh` (the `using` behavior of C++11 would be handy here).

**Comment by lrineau** [comment]
The tested template [`CGAL::Surface_mesh<P>::Property_map` is documented][doc].

[doc]: http://doc.cgal.org/latest/Surface_mesh/classCGAL_1_1Surface__mesh.html#a9d40ac48b19106cd3f59639b036471b1

This PR cannot change the API without at least a small feature, and anyway changes of API without backward -compatibility are highly discouraged.

In C++11, the `using` is a perfect solution. In C++<=03, could `Surface_mesh<P>::Property_map<I,T>` derive from `CGAL::Properties::Property_map<I,T>`?

**Comment by sgiraudot** [comment]
Yes, that was the solution is used inside the `Point_set_3` structure but it requires more work (additional manual casts).

So probably forget about this PR right now, I'll have to spend some time on this before it's ready.

**Comment by lrineau** [comment]
I think I have fixed the compatibility issue with 22da63879822a6f715acfa48be838759e8e4586e. I have remerged the PR into `integration`.

**Comment by lrineau** [comment]
Now my build stops in the compilation of the demo:
```
/cgal/Polyhedron/demo/Polyhedron/./include/CGAL/Point_set_3.h: In instantiation of ‘CGAL::Point_set_3<Point, Vector>::iterator CGAL::Point_set_3<Point, Vector>::insert(const Point&, const Vector&) [with Point = CGAL::Point_3<CGAL::Epick>; Vector = CGAL::Vector_3<CGAL::Epick>; CGAL::Point_set_3<Point, Vector>::iterator = __gnu_cxx::__normal_iterator<CGAL::Point_set_3<CGAL::Point_3<CGAL::Epick>, CGAL::Vector_3<CGAL::Epick> >::Index*, std::vector<CGAL::Point_set_3<CGAL::Point_3<CGAL::Epick>, CGAL::Vector_3<CGAL::Epick> >::Index, std::allocator<CGAL::Point_set_3<CGAL::Point_3<CGAL::Epick>, CGAL::Vector_3<CGAL::Epick> >::Index> > >]’:
/cgal/Polyhedron/demo/Polyhedron/Scene_points_with_normal_item.cpp:73:27:   required from here
/cgal/Polyhedron/demo/Polyhedron/./include/CGAL/Point_set_3.h:349:24: error: ‘has_normals’ was not declared in this scope
     assert (has_normals());
                        ^
```

**Comment by sgiraudot** [comment]
Arf, sorry, I don't know how I did not spot this error earlier… I'm correcting it right now.

**Comment by lrineau** [comment]
Compilation errors with (probably) older versions of Boost:
```
In file included from .../test/Polyhedron_Demo/./include/Point_set_3.h:28:0,
                 from .../test/Polyhedron_Demo/Scene_points_with_normal_item.h:7,
                 from .../test/Polyhedron_Demo/Scene_points_with_normal_item.cpp:1:
.../test/Polyhedron_Demo/./include/CGAL/Point_set_3.h: In member function 'std::string CGAL::Point_set_3<Point, Vector>::properties() const':
.../test/Polyhedron_Demo/./include/CGAL/Point_set_3.h:750:43: error: 'boost::core' has not been declared
     oss << "CGAL::Point_set_3<" << boost::core::demangle(typeid(Point).name())
                                           ^
../test/Polyhedron_Demo/./include/CGAL/Point_set_3.h:756:21: error: 'boost::core' has not been declared
           << boost::core::demangle(m_base.get_type(prop[i]).name()) << std::endl;
                     ^
```
https://cgal.geometryfactory.com/CGAL/testsuite/results-4.10-Ic-20.shtml#Polyhedron_Demo

**Comment by lrineau** [comment]
All the compilation errors are:
```
cgal /srv/CGAL/www/testsuite/CGAL-4.10-Ic-20 $ zgrep --no-filename -E '(error): ' Polyhedron_Demo/*gz | sort -u
E:\CGAL\test\CGAL-4.10-Ic-20\cmake\platforms\x64_Cygwin-Windows10_MSVC2015-Release-64bits\test\Polyhedron_Demo\.\include\CGAL/Point_set_3.h(144): error C2061: syntax error: identifier 'const_iterator'
/home/cgal_tester/build/src/cmake/platforms/ArchLinux-clang/test/Polyhedron_Demo/./include/CGAL/IO/read_ply_point_set_3.h:71:42: error: 'Push_property_map' following the 'template' keyword does not refer to a template
/home/cgal_tester/build/src/cmake/platforms/CentOS6-CXX11-Boost157/test/Polyhedron_Demo/Scene_points_with_normal_item.cpp:748: error: using ‘typename’ outside of template
/home/cgal_tester/build/src/cmake/platforms/CentOS6/test/Polyhedron_Demo/./include/CGAL/Point_set_3.h:750: error: ‘boost::core’ has not been declared
/home/cgal_tester/build/src/cmake/platforms/CentOS6/test/Polyhedron_Demo/./include/CGAL/Point_set_3.h:756: error: ‘boost::core’ has not been declared
/home/cgal_tester/build/src/cmake/platforms/CentOS6/test/Polyhedron_Demo/Scene_points_with_normal_item.cpp:748: error: using ‘typename’ outside of template
/home/cgal_tester/build/src/cmake/platforms/CentOS7-ICC-Release/test/Polyhedron_Demo/./include/CGAL/Point_set_3.h(750): error: name followed by "::" must be a class or namespace name
/home/cgal_tester/build/src/cmake/platforms/CentOS7-ICC-Release/test/Polyhedron_Demo/./include/CGAL/Point_set_3.h(756): error: name followed by "::" must be a class or namespace name
/home/cgal_tester/build/src/cmake/platforms/CentOS7/test/Polyhedron_Demo/./include/CGAL/Point_set_3.h:750:43: error: 'boost::core' has not been declared
/home/cgal_tester/build/src/cmake/platforms/CentOS7/test/Polyhedron_Demo/./include/CGAL/Point_set_3.h:756:21: error: 'boost::core' has not been declared
/home/cgal_tester/build/src/cmake/platforms/Ubuntu-clang-latest/test/Polyhedron_Demo/./include/CGAL/IO/read_ply_point_set_3.h:71:42: error: 'Push_property_map' following the 'template' keyword does not refer to a template
/Users/cgaltester/cgal_test/CGAL-4.10-Ic-20/cmake/platforms/x86-64_Darwin-13.0_Apple-clang-5.0_Release-cpp11/test/Polyhedron_Demo/./include/CGAL/Point_set_3.h:750:43: error: no member named 'core' in namespace 'boost'
/Users/cgaltester/cgal_test/CGAL-4.10-Ic-20/cmake/platforms/x86-64_Darwin-13.0_Apple-clang-5.0_Release-cpp11/test/Polyhedron_Demo/./include/CGAL/Point_set_3.h:756:21: error: no member named 'core' in namespace 'boost'
/Users/cgaltester/cgal_test/CGAL-4.10-Ic-20/cmake/platforms/x86-64_Darwin-13.0_Apple-clang-5.0_Release/test/Polyhedron_Demo/./include/CGAL/Point_set_3.h:750:43: error: no member named 'core' in namespace 'boost'
/Users/cgaltester/cgal_test/CGAL-4.10-Ic-20/cmake/platforms/x86-64_Darwin-13.0_Apple-clang-5.0_Release/test/Polyhedron_Demo/./include/CGAL/Point_set_3.h:756:21: error: no member named 'core' in namespace 'boost'
```

**Comment by lrineau** [comment]
And warnings related to the point set item:
```
cgal /srv/CGAL/www/testsuite/CGAL-4.10-Ic-20 $ zgrep --no-filename -E '(warning): ' Polyhedron_Demo/*gz | sort -u  | grep -i -E 'point.*:[0-9]' 
/home/cgal_tester/build/src/cmake/platforms/ArchLinux-clang/test/Polyhedron_Demo/./include/CGAL/Point_set_3.h:847:60: warning: 'const' qualifier on reference type 'reference' (aka 'typename Property::value_type &') has no effect [-Wignored-qualifiers]
/home/cgal_tester/build/src/cmake/platforms/ArchLinux-clang/test/Polyhedron_Demo/./include/CGAL/Point_set_3.h:856:12: warning: 'const' qualifier on reference type 'reference' (aka 'typename Property::value_type &') has no effect [-Wignored-qualifiers]
/home/cgal_tester/build/src/cmake/platforms/ArchLinux-clang/test/Polyhedron_Demo/./include/CGAL/Point_set_3.h:95:29: warning: class template 'Property_map' was previously declared as a struct template [-Wmismatched-tags]
/home/cgal_tester/build/src/cmake/platforms/Ubuntu-clang-latest/test/Polyhedron_Demo/./include/CGAL/Point_set_3.h:847:60: warning: 'const' qualifier on reference type 'reference' (aka 'typename Property::value_type &') has no effect [-Wignored-qualifiers]
/home/cgal_tester/build/src/cmake/platforms/Ubuntu-clang-latest/test/Polyhedron_Demo/./include/CGAL/Point_set_3.h:856:12: warning: 'const' qualifier on reference type 'reference' (aka 'typename Property::value_type &') has no effect [-Wignored-qualifiers]
/home/cgal_tester/build/src/cmake/platforms/Ubuntu-clang-latest/test/Polyhedron_Demo/Scene_points_with_normal_item.cpp:748:13: warning: 'typename' occurs outside of a template [-Wc++11-extensions]
/no-raid/home/lrineau/CGAL/CGAL-4.10-Ic-20/cmake/platforms/x86-64_Linux-Fedora19_llvm-clang-with-g++-4.8/test/Polyhedron_Demo/./include/CGAL/Point_set_3.h:847:60: warning: 'const' qualifier on reference type 'reference' (aka 'typename Property::value_type &') has no effect [-Wignored-qualifiers]
/no-raid/home/lrineau/CGAL/CGAL-4.10-Ic-20/cmake/platforms/x86-64_Linux-Fedora19_llvm-clang-with-g++-4.8/test/Polyhedron_Demo/./include/CGAL/Point_set_3.h:856:12: warning: 'const' qualifier on reference type 'reference' (aka 'typename Property::value_type &') has no effect [-Wignored-qualifiers]
/no-raid/home/lrineau/CGAL/CGAL-4.10-Ic-20/cmake/platforms/x86-64_Linux-Fedora19_llvm-clang-with-g++-4.8/test/Polyhedron_Demo/./include/CGAL/Point_set_3.h:95:29: warning: class template 'Property_map' was previously declared as a struct template [-Wmismatched-tags]
/no-raid/home/lrineau/CGAL/CGAL-4.10-Ic-20/cmake/platforms/x86-64_Linux-Fedora19_llvm-clang-with-g++-4.8/test/Polyhedron_Demo/Scene_points_with_normal_item.cpp:748:13: warning: 'typename' occurs outside of a template [-Wc++11-extensions]
/Users/cgaltester/cgal_test/CGAL-4.10-Ic-20/cmake/platforms/x86-64_Darwin-13.0_Apple-clang-5.0_Release-cpp11/test/Polyhedron_Demo/./include/CGAL/Point_set_3.h:847:60: warning: 'const' qualifier on reference type 'reference' (aka 'typename Property::value_type &') has no effect [-Wignored-qualifiers]
/Users/cgaltester/cgal_test/CGAL-4.10-Ic-20/cmake/platforms/x86-64_Darwin-13.0_Apple-clang-5.0_Release-cpp11/test/Polyhedron_Demo/./include/CGAL/Point_set_3.h:856:12: warning: 'const' qualifier on reference type 'reference' (aka 'typename Property::value_type &') has no effect [-Wignored-qualifiers]
/Users/cgaltester/cgal_test/CGAL-4.10-Ic-20/cmake/platforms/x86-64_Darwin-13.0_Apple-clang-5.0_Release-cpp11/test/Polyhedron_Demo/./include/CGAL/Point_set_3.h:95:29: warning: class template 'Property_map' was previously declared as a struct template [-Wmismatched-tags]
/Users/cgaltester/cgal_test/CGAL-4.10-Ic-20/cmake/platforms/x86-64_Darwin-13.0_Apple-clang-5.0_Release/test/Polyhedron_Demo/./include/CGAL/Point_set_3.h:847:60: warning: 'const' qualifier on reference type 'reference' (aka 'typename Property::value_type &') has no effect [-Wignored-qualifiers]
/Users/cgaltester/cgal_test/CGAL-4.10-Ic-20/cmake/platforms/x86-64_Darwin-13.0_Apple-clang-5.0_Release/test/Polyhedron_Demo/./include/CGAL/Point_set_3.h:856:12: warning: 'const' qualifier on reference type 'reference' (aka 'typename Property::value_type &') has no effect [-Wignored-qualifiers]
/Users/cgaltester/cgal_test/CGAL-4.10-Ic-20/cmake/platforms/x86-64_Darwin-13.0_Apple-clang-5.0_Release/test/Polyhedron_Demo/./include/CGAL/Point_set_3.h:95:29: warning: class template 'Property_map' was previously declared as a struct template [-Wmismatched-tags]
/Users/cgaltester/cgal_test/CGAL-4.10-Ic-20/cmake/platforms/x86-64_Darwin-13.0_Apple-clang-5.0_Release/test/Polyhedron_Demo/Scene_points_with_normal_item.cpp:748:13: warning: 'typename' occurs outside of a template [-Wc++11-extensions]
```

**Comment by lrineau** [comment]
Simon, can you please merge again this PR into `integration`? Be sure to fetch/pull `cgal-dev/integration` before, because I have modified it very recently.

**Comment by sgiraudot** [comment]
Okay, it's done.

**Comment by sgiraudot** [comment]
It seems like my `boost::demangle` has broken everywhere... I'll look into it.

**Comment by sgiraudot** [comment]
I fixed it and merged it back into `integration`.

**Comment by sgiraudot** [comment]
I fixed the warning about conversions (`std::size_t` to `int`).

The error about the `template` keyword we get with clang is weird: I tried to compile the demo with clang on my machine _and_ on a docker container and got no error… I have no idea how to debug it.

**Comment by lrineau** [comment]
@sgiraudot Now that I have merged a lot in `master`, would you mind rebasing this branch/PR, to fix the merge conflict?

**Comment by lrineau** [comment]
I can reproduce the strange compilation error with the clang++-3.8.0 I have on my machine:
```
clang version 3.8.0 (tags/RELEASE_380/final)
```

With the following patch it does compile:
```patch
diff --git a/Polyhedron/demo/Polyhedron/include/CGAL/Point_set_3.h b/Polyhedron/demo/Polyhedron/include/CGAL/Point_set_3.h
index 9028232..1e94b4b 100644
--- a/Polyhedron/demo/Polyhedron/include/CGAL/Point_set_3.h
+++ b/Polyhedron/demo/Polyhedron/include/CGAL/Point_set_3.h
@@ -78,8 +78,8 @@ public:
   };
   typedef Property_map<Index> Index_map;
 
-  template <typename Property>
-  class Push_property_map;
+  // template <typename Property>
+  // class Push_property_map;
   /// \endcond
   
   /*!
@@ -95,7 +95,7 @@ public:
     friend class Properties::Property_container<Index>;
     template <class> friend class Properties::Property_array;
     template <class> friend struct Property_map;
-    template <class> friend class Push_property_map;
+    //    template <class> friend class Push_property_map;
     friend class std::vector<Index>;
     std::size_t value;
     
```

Why does `Index` need to declare so many friends?

**Comment by sgiraudot** [comment]
This was done to prevent the user from doing something like `point_set.point(Index(12));`: the indices should only be instantiated by allowed classes (the constructor that takes a `std::size_t` parameter is private, hence the friend classes).

It does not seem necessary anymore for `Push_property_map` which never creates a new index from scratch, so your patch looks good for me.

**Comment by lrineau** [comment]
I let you apply the patch, along with the rebasing on `cgal/master`.

**Comment by sgiraudot** [comment]
Okay, it is up to date with master and your fix. I compiled the demo locally, no error or warning.

**Comment by lrineau** [comment]
This PR is now in conflict with #1565. :-(

**Comment by sgiraudot** [comment]
I fixed the conflict, it's merged into integration.

**Comment by sgiraudot** [comment]
Apparently my conflict resolution was not good and all tests with affine transform failed… I'll look into it.

**Comment by sgiraudot** [comment]
I merged it back into integration with a correct conflict resolution, compiled without error locally.

**Comment by lrineau** [comment]
A new bunch of PRs have been merged into `master` today. There is still a conflict between your PR (this #1527) and `master`, in the file `Polyhedron/demo/Polyhedron/Plugins/PCA/Affine_transform_plugin.cpp`. I suggest you rebase again, hopefully for the last time.

**Comment by sgiraudot** [comment]
The branch is up to date. Unfortunately `git rebase` did not seem to work well this time, so I had to rebase from `master` using the new branch + merge method (hence the parasite commits that now appear in this PR). Sorry about that :(

**Comment by lrineau** [comment]
It seems your have renewed your branch from `cgal-dev/integration` (5aaa9c0) instead of `cgal/master`.

Use this command:
```
git rebase 5aaa9c0 --onto cgal/master
```
to get a proper rebased branch from `cgall/master`.

**Comment by sgiraudot** [comment]
I couldn't make `rebase` work (some files were missing and I don't understand why) but I managed to get a clean branch by cherry-picking the necessary commits on a clean branch based on `cgal/master`.

**Comment by lrineau** [comment]
That's strange the rebase did not work for you, while I ran it correctly on my machine yesterday.

Anyway, with the series of cherry-picks, you have implemented the rebase by hand. That produces the same expected result. I will merge the PR in `integration` for testing tonight.

**Comment by sgiraudot** [comment]
I saw that you already pushed it into `integration`, but as I have added another commit since then, I pushed it again (the commit is no big deal, just an update from the file from `Point_set_3-GF` branch, these are mostly comments + a method that had a change of name).

File: Polyhedron/demo/Polyhedron/Scene_surface_mesh_item.cpp [hunk 1]
```diff
@@ -57,8 +57,8 @@ struct Scene_surface_mesh_item_priv{
   //! fill the flat data vectors.
   void
   triangulate_facet(face_descriptor fd,
-                    SMesh::Property_map<face_descriptor, Kernel::Vector_3 > *fnormals,
-                    SMesh::Property_map<face_descriptor, CGAL::Color> *fcolors,
+                    CGAL::Properties::Property_map<face_descriptor, Kernel::Vector_3 > *fnormals,
+                    CGAL::Properties::Property_map<face_descriptor, CGAL::Color> *fcolors,
```
Comment (lrineau) [inline Polyhedron/demo/Polyhedron/Scene_surface_mesh_item.cpp]: @sgiraudot Note for another time: that kind of required change in the code of the demo was an indication that the API was broken by the PR.

Cross-reference: PR #1890 is dedicated to fixing this, and restore the proper API.

**Pull request merged**